src/library/dao/trackdao.cpp

```diff
@@ -628,49 +628,53 @@ bool TrackDAO::addTracksAdd(TrackInfoObject* pTrack, bool unremove) {
             LOG_FAILED_QUERY(*m_pQueryLibraryInsert)
                     << "Failed to INSERT new track into library:"
                     << pTrack->getLocation();
-            return false;
+            return TrackId();
         }
         trackId = TrackId(m_pQueryLibraryInsert->lastInsertId());
-        pTrack->setId(trackId);
-        m_analysisDao.saveTrackAnalyses(pTrack);
-        m_cueDao.saveTrackCues(trackId, pTrack);
+        m_analysisDao.saveTrackAnalyses(pTrack.data());
+        m_cueDao.saveTrackCues(trackId, pTrack.data());
         pTrack->resetDirty();
+        m_tracksAddedSet.insert(trackId);
     }
-    m_tracksAddedSet.insert(trackId);
-    return true;
+    if (trackId.isValid()) {
+        pTrack->setId(trackId);
+    } else {
+        qWarning() << "TrackDAO::addTracksAdd:"
+                << "Failed to add track to database"
+                << pTrack->getLocation();
+    }
+    DEBUG_ASSERT(pTrack->getId() == trackId);
```

Comment:
this Assert here is hard to understand some comments about the possible cases would be useful.